Error while calculating TTD ETA

### Description
Division by 0 error while estimating the time until TTD:

```
{"timestamp":"2022-05-03T03:24:05,812","level":"ERROR","thread":"beaconchain-async-4","class":"TerminalPowBlockMonitor","message":"Unexpected error while checking TTD","throwable":" java.util.concurrent.CompletionException: java.lang.ArithmeticException: divide by zero
at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1159)
at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:147)
at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$13(SafeFuture.java:269)
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:81)
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:38)
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:73)
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)\nCaused by: java.lang.ArithmeticException: divide by zero
at org.apache.tuweni.units.bigints.UInt256.divide(UInt256.java:425)
at tech.pegasys.teku.statetransition.forkchoice.TerminalPowBlockMonitor.checkTtdEta(TerminalPowBlockMonitor.java:300)
at tech.pegasys.teku.statetransition.forkchoice.TerminalPowBlockMonitor.lambda$checkTerminalBlockByTTD$6(TerminalPowBlockMonitor.java:237)
at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)\n\t... 15 more\n"}

```

--- a/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
+++ b/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
@@ -293,15 +293,18 @@ public class TerminalPowBlockMonitor {
                 .orElse(UInt256.ZERO)
                 .divide(lastTotalDifficultyDiffs.size());
         final UInt256 averageDifficultyPerSecond = diffAverage.divide(pollingPeriod.getSeconds());
-
         final UInt256 ttd = specConfigBellatrix.getTerminalTotalDifficulty();
 
-        final UInt256 secondsToTTD =
-            ttd.subtract(totalDifficulty).divide(averageDifficultyPerSecond);
+        if (averageDifficultyPerSecond.isZero()) {
+          eventLogger.terminalPowBlockTtdEta(ttd, Optional.empty());
+        } else {
+          final UInt256 secondsToTTD =
+              ttd.subtract(totalDifficulty).divide(averageDifficultyPerSecond);
 
-        final Duration eta = Duration.ofSeconds(secondsToTTD.toLong());
+          final Duration eta = Duration.ofSeconds(secondsToTTD.toLong());
 
-        eventLogger.terminalPowBlockTtdEta(ttd, eta);
+          eventLogger.terminalPowBlockTtdEta(ttd, Optional.of(eta));
+        }
       }
     }
 

--- a/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
+++ b/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
@@ -295,15 +295,13 @@ public class TerminalPowBlockMonitor {
         final UInt256 averageDifficultyPerSecond = diffAverage.divide(pollingPeriod.getSeconds());
         final UInt256 ttd = specConfigBellatrix.getTerminalTotalDifficulty();
 
-        if (averageDifficultyPerSecond.isZero()) {
-          eventLogger.terminalPowBlockTtdEta(ttd, Optional.empty());
-        } else {
+        if (!averageDifficultyPerSecond.isZero()) {
           final UInt256 secondsToTTD =
               ttd.subtract(totalDifficulty).divide(averageDifficultyPerSecond);
 
           final Duration eta = Duration.ofSeconds(secondsToTTD.toLong());
 
-          eventLogger.terminalPowBlockTtdEta(ttd, Optional.of(eta));
+          eventLogger.terminalPowBlockTtdEta(totalDifficulty, eta);
         }
       }
     }

--- a/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
+++ b/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
@@ -39,7 +39,7 @@ public class TerminalPowBlockMonitor {
   private static final Logger LOG = LogManager.getLogger();
   // number of samples to average out totalDifficulty
   private static final int TD_DIFF_SAMPLES = 5;
-  // how many times
+  // how many times we produce the event, based on polling period (secondsPerEth1Block)
   private static final int ETA_EVENT_FREQUENCY_IN_POLLING_PERIODS = 5;
 
   private final EventLogger eventLogger;

--- a/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
+++ b/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
@@ -460,15 +460,15 @@ public class TerminalPowBlockMonitorTest {
     pollTtd(UInt256.ZERO);
     pollTtd(UInt256.ZERO);
 
-    verify(eventLogger).terminalPowBlockTtdEta(TTD, Optional.empty());
-
     pollTtd(TTD.subtract(tdDiff.multiply(5)));
     pollTtd(TTD.subtract(tdDiff.multiply(4)));
     pollTtd(TTD.subtract(tdDiff.multiply(3)));
     pollTtd(TTD.subtract(tdDiff.multiply(2)));
-    pollTtd(TTD.subtract(tdDiff.multiply(1)));
 
-    verify(eventLogger).terminalPowBlockTtdEta(TTD, Optional.of(Duration.ofSeconds(17)));
+    final UInt256 expectedEventTD = TTD.subtract(tdDiff.multiply(1));
+    pollTtd(expectedEventTD);
+
+    verify(eventLogger).terminalPowBlockTtdEta(expectedEventTD, Duration.ofSeconds(17));
 
     verifyNoMoreInteractions(eventLogger);
   }

--- a/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
+++ b/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
@@ -452,10 +452,15 @@ public class TerminalPowBlockMonitorTest {
 
     terminalPowBlockMonitor.start();
 
-    final UInt64 bellatrixSlot =
-        BELLATRIX_FORK_EPOCH.times(spec.getGenesisSpecConfig().getSlotsPerEpoch());
-
     goToSlot(BELLATRIX_FORK_EPOCH.times(spec.getGenesisSpecConfig().getSlotsPerEpoch()));
+
+    pollTtd(UInt256.ZERO);
+    pollTtd(UInt256.ZERO);
+    pollTtd(UInt256.ZERO);
+    pollTtd(UInt256.ZERO);
+    pollTtd(UInt256.ZERO);
+
+    verify(eventLogger).terminalPowBlockTtdEta(TTD, Optional.empty());
 
     pollTtd(TTD.subtract(tdDiff.multiply(5)));
     pollTtd(TTD.subtract(tdDiff.multiply(4)));
@@ -463,9 +468,8 @@ public class TerminalPowBlockMonitorTest {
     pollTtd(TTD.subtract(tdDiff.multiply(2)));
     pollTtd(TTD.subtract(tdDiff.multiply(1)));
 
-    verify(eventLogger)
-        .terminalPowBlockTtdEta(
-            TTD, Duration.ofSeconds(spec.getSecondsPerEth1Block(bellatrixSlot)));
+    verify(eventLogger).terminalPowBlockTtdEta(TTD, Optional.of(Duration.ofSeconds(17)));
+
     verifyNoMoreInteractions(eventLogger);
   }
 

--- a/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
+++ b/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
@@ -204,7 +204,7 @@ public class EventLogger {
             eta.toMinutes() - TimeUnit.HOURS.toMinutes(eta.toHours()),
             eta.getSeconds() - TimeUnit.MINUTES.toSeconds(eta.toMinutes()));
 
-    log.info(String.format("Current Total Difficulty: %s - TTD ETA: %s", ttd, etaString));
+    log.info(String.format("TTD ETA: %s - Current Total Difficulty: %s", etaString, ttd));
   }
 
   public void transitionConfigurationTtdTbhMismatch(

--- a/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
+++ b/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
@@ -19,6 +19,7 @@ import java.time.Duration;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -185,15 +186,19 @@ public class EventLogger {
         Color.GREEN);
   }
 
-  public void terminalPowBlockTtdEta(final UInt256 ttd, final Duration eta) {
+  public void terminalPowBlockTtdEta(final UInt256 ttd, final Optional<Duration> maybeEta) {
 
     final String etaString =
-        String.format(
-            "%s days and %sh %sm %ss",
-            eta.toDays(),
-            eta.toHours() - TimeUnit.DAYS.toHours(eta.toDays()),
-            eta.toMinutes() - TimeUnit.HOURS.toMinutes(eta.toHours()),
-            eta.getSeconds() - TimeUnit.MINUTES.toSeconds(eta.toMinutes()));
+        maybeEta
+            .map(
+                eta ->
+                    String.format(
+                        "%s days and %sh %sm %ss",
+                        eta.toDays(),
+                        eta.toHours() - TimeUnit.DAYS.toHours(eta.toDays()),
+                        eta.toMinutes() - TimeUnit.HOURS.toMinutes(eta.toHours()),
+                        eta.getSeconds() - TimeUnit.MINUTES.toSeconds(eta.toMinutes())))
+            .orElse("N/A");
 
     log.info(String.format("TTD (%s) ETA: %s", ttd, etaString));
   }

--- a/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
+++ b/infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java
@@ -19,7 +19,6 @@ import java.time.Duration;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -195,21 +194,17 @@ public class EventLogger {
         Color.GREEN);
   }
 
-  public void terminalPowBlockTtdEta(final UInt256 ttd, final Optional<Duration> maybeEta) {
+  public void terminalPowBlockTtdEta(final UInt256 ttd, final Duration eta) {
 
     final String etaString =
-        maybeEta
-            .map(
-                eta ->
-                    String.format(
-                        "%s days and %sh %sm %ss",
-                        eta.toDays(),
-                        eta.toHours() - TimeUnit.DAYS.toHours(eta.toDays()),
-                        eta.toMinutes() - TimeUnit.HOURS.toMinutes(eta.toHours()),
-                        eta.getSeconds() - TimeUnit.MINUTES.toSeconds(eta.toMinutes())))
-            .orElse("N/A");
+        String.format(
+            "%s days and %sh %sm %ss",
+            eta.toDays(),
+            eta.toHours() - TimeUnit.DAYS.toHours(eta.toDays()),
+            eta.toMinutes() - TimeUnit.HOURS.toMinutes(eta.toHours()),
+            eta.getSeconds() - TimeUnit.MINUTES.toSeconds(eta.toMinutes()));
 
-    log.info(String.format("TTD (%s) ETA: %s", ttd, etaString));
+    log.info(String.format("Current Total Difficulty: %s - TTD ETA: %s", ttd, etaString));
   }
 
   public void transitionConfigurationTtdTbhMismatch(

